serve/mlc_serve/model/model_common.py

```diff
@@ -85,21 +83,17 @@ def sample_from_logits(
     requests: Sequence[RequestType],
     sampling_metadata: SamplingState,
     vocab_size: int,
-    copy_stream: torch.cuda.Stream,
     torch_dtype: torch.dtype,
     torch_dev: str,
     past_decode_tokens: List[List[int]],
 ) -> List[TextGenerationResult]:
     batch_size = logits.shape[0]
     assert batch_size == len(requests)
+
     # Convert to torch tensors if logits are in tvm ndarray
     if isinstance(logits, tvm.nd.NDArray):
         logits = torch.from_dlpack(logits)
 
-    # synchronization point for sampling tensors
-    # wait until all the tensors are loaded on GPU
-    torch.cuda.current_stream().wait_stream(copy_stream)
```

Comment:
To simplify for the PT path, I changed `wait_stream` to be done before `sample_from_logits` is called.